### PR TITLE
Enable Istio authorization policy for the validator and the HTTP source adapter

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Enable Istio authorization policy for the validator and the HTTP source adapter.

**Related issue(s)**
- https://github.com/kyma-project/kyma/issues/7955

### CODE REVIEW

File: components/application-operator/charts/application/templates/authorization-policy.yaml [hunk 1]
```diff
@@ -60,6 +60,7 @@ spec:
     - from:
         - source:
             principals:
+              - cluster.local/ns/e2e-upgrade-test/sa/test-e2e-upgrade-test
```
Comment (k15r) [inline components/application-operator/charts/application/templates/authorization-policy.yaml:63]: i know that this is necessary with the current implementation of the test, but i really dont like that we reference test resources in a production component. Isn't there another solution?
e.g. Use one of the production service accounts for the test?

Comment (marcobebway) [reply components/application-operator/charts/application/templates/authorization-policy.yaml]: Yes, you're correct. I created an authorization policy for the upgrade test under its charts.

**Pull request merged**